src/org/zaproxy/zap/extension/ascanrulesAlpha/TestUserAgent.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+/**
+ *  @author kniepdennis@gmail.com
+ */
+
+public class TestUserAgent extends AbstractAppPlugin {
+
+    private static Logger log = Logger.getLogger(TestUserAgent.class);
+
+    private static final int PLUGIN_ID = 30897;
+
+    private static final String InternetExplorer8 = "Mozilla/4.0 (compatible; MSIE 8.0; Windows NT 6.1)";
+    private static final String InternetExplorer7 = "Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 6.0)";
+    private static final String InternetExplorer6 = "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)";
+    private static final String GoogleBot2_1 = "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)";
+    private static final String MsnBot1_1 = "msnbot/1.1 (+http://search.msn.com/msnbot.htm)";
+    private static final String YahooSlurp = "Mozilla/5.0 (compatible; Yahoo! Slurp; http://help.yahoo.com/help/us/ysearch/slurp)";
+    private static final String iPhone3 = "Mozilla/5.0 (iPhone; U; CPU iPhone OS 3_0 like Mac OS X; en-us) AppleWebKit/528.18 (KHTML, like Gecko) Version/4.0 Mobile/7A341 Safari/528.16";
+
+    private static final String[] USER_AGENTS = {
+            InternetExplorer8,
+            InternetExplorer7,
+            InternetExplorer6,
+            GoogleBot2_1,
+            MsnBot1_1,
+            YahooSlurp,
+            iPhone3
+    };
+
+    @Override
+    public int getId() {
+        return PLUGIN_ID;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString("ascanalpha.testuseragent.name");
```

Comment:
Consider extracting the segment "ascanalpha." to a constant as it will make the migrations to beta/master easier (i.e. the string just needs to be updated in one place).